src/wallet/test/wallet_test_fixture.cpp

```diff
@@ -82,16 +82,18 @@ TestChain100Setup::TestChain100Setup(UnitEInjectorConfiguration config)
 // scriptPubKey, and try to add it to the current chain.
 //
 CBlock
-TestChain100Setup::CreateAndProcessBlock(const std::vector<CMutableTransaction>& txns, const CScript& coinbase_script, bool *processed)
+TestChain100Setup::CreateAndProcessBlock(const std::vector<CMutableTransaction>& txns,
+                                         const CScript& coinbase_script,
+                                         boost::optional<staking::Coin> stake,
+                                         bool *processed)
 {
   const CChainParams& chainparams = Params();
 
   esperanza::WalletExtension &wallet_ext = m_wallet->GetWalletExtension();
 
-  staking::CoinSet coins;
-  {
+  if (!stake) {
     LOCK2(m_active_chain->GetLock(), wallet_ext.GetLock());
-    coins = wallet_ext.GetStakeableCoins();
+    stake.emplace(*wallet_ext.GetStakeableCoins().begin());
```

Comment:
could it happen that `GetStakeableCoins` does not return any coins here, i.e. `begin() == end()`?

Reply:
I will add check